sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -326,12 +326,24 @@ def _resolve_output_path_placeholder(output_key: str) -> str:
       pipeline_params = _pipeline_param.extract_pipelineparams_from_any(
           argument_value)
       if pipeline_params and is_compiling_for_v2:
-        # argument_value contains PipelineParam placeholders.
-        raise NotImplementedError(
-            'Currently, a component input can only accept either a constant '
-            'value or a reference to another pipeline parameter. It cannot be a '
-            'combination of both. Got: {} for input {}'.format(
-                argument_value, input_name))
+        # argument_value contains PipelineParam placeholders which needs to be
+        # replaced. And the input needs to be added to the task spec.
+        for param in pipeline_params:
+          additional_input_name = '{}-{}'.format(input_name, param.full_name)
```

Comment:
Is this format for parameter name by convention? Should this be a shared method instead?

Reply:
The naming here is only for one special scenario -- compiler injects an input parameter, which so far only happens in this specific code location. i.e. if it's extracted into a shared method, that method will only be called here. So probably not necessarily to make a method for this?

That said, there're two requirements for the naming conversion:
1. The name needs to be unique among all injected inputs for the current component. (This is guaranteed because `param.full_name` is unique)
2. The name also shouldn't collide with any existing user defined input for the current component.

I added some comments in the code, and changed to use double dashes to form the name, so hopefully it should rarely collide with user defined inputs. I also added a check, in case name collision happens, we at least throw an error here.

Of course, we can have some logic to iterate through all the existing input names, and always generate a unique name. But I feel it's a bit overkill, with the double dash and collision check, it should be good for most cases. WDYT?